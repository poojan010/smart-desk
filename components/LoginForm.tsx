import React, { useState } from 'react';
import { View, TextInput, Button, StyleSheet } from 'react-native';

interface LoginFormProps {
  onSubmit: (credentials: { email: string; password: string }) => void;
}

const LoginForm: React.FC<LoginFormProps> = ({ onSubmit }) => {
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');

  return (
    <View style={styles.container}>
      <TextInput
        value={email}
        onChangeText={setEmail}
        placeholder="Email"
        keyboardType="email-address"
        style={styles.input}
      />
      <TextInput
        value={password}
        onChangeText={setPassword}
        placeholder="Password"
        secureTextEntry
        style={styles.input}
      />
      <Button title="Submit" onPress={() => onSubmit({ email, password })} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
  },
  input: {
    height: 40,
    borderColor: '#ccc',
    borderWidth: 1,
    marginBottom: 10,
    paddingHorizontal: 10,
  },
});

export default LoginForm;
