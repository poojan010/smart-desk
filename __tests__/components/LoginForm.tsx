import { act, render, screen, userEvent } from "@testing-library/react-native";

import LoginForm from "@/components/LoginForm";

describe("<LoginForm/>", () => {

    // test("Snapshot", () => {
    //     const onSubmitMock = jest.fn()
    //     const { toJSON } = render(<LoginForm onSubmit={onSubmitMock} />)

    //     expect(toJSON()).toMatchSnapshot()
    // })

    test("LoginForm renders correctly", () => {
        const onSubmitMock = jest.fn()
        render(<LoginForm onSubmit={onSubmitMock} />)

        screen.getByTestId("email-input")
        screen.getByTestId("password-input")
        screen.getByTestId("submit-button")
    })

    test("User can enter values into input fields",async () => {
        const onSubmitMock = jest.fn()
        render(<LoginForm onSubmit={onSubmitMock} />)

        const user = userEvent.setup();

        const emailInput = screen.getByTestId("email-input")
        const passwordInput = screen.getByTestId("password-input")

        const emailValue = "abcd@test.com"
        const passwordValue = "password123"

        await act(async () => {
            await user.type(emailInput,emailValue)
        });
        
        await act(async () => {
            await user.type(passwordInput,passwordValue)
        });
        
        expect(emailInput.props.value).toBe(emailValue);
        expect(passwordInput.props.value).toBe(passwordValue);
    })

    test("onSubmit is called with correct values when submitting the form",async () => {
        const onSubmitMock = jest.fn()

        render(<LoginForm onSubmit={onSubmitMock} />)

        const user = userEvent.setup();

        const emailInput = screen.getByTestId("email-input")
        const passwordInput = screen.getByTestId("password-input")
        const button = screen.getByTestId("submit-button")

        const emailValue = "abcd@test.com"
        const passwordValue = "password123"

        await act(async () => {
            await user.type(emailInput,emailValue)
        });
        
        await act(async () => {
            await user.type(passwordInput,passwordValue)
        });
        
        await act(async () => {
            await user.press(button)
        });

        expect(onSubmitMock).toHaveBeenCalledWith({
            email: emailValue,
            password: passwordValue
        })
    })

})