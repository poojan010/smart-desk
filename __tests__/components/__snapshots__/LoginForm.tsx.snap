// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm/> Snapshot 1`] = `
<View
  style={
    {
      "padding": 10,
    }
  }
>
  <TextInput
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Email"
    style={
      {
        "borderColor": "#ccc",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "paddingHorizontal": 10,
      }
    }
    testID="email-input"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      {
        "borderColor": "#ccc",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "paddingHorizontal": 10,
      }
    }
    testID="password-input"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="submit-button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
